diff --git a/dist/encryption.js b/dist/encryption.js
index b68674b1cb7754bc8451e72287a7541e736546c5..35d5d8d6ee53f60d940e5ff780954fb8da95f1f2 100644
--- a/dist/encryption.js
+++ b/dist/encryption.js
@@ -116,11 +116,11 @@ function encryptSafely({ publicKey, data, version, }) {
     };
     // calculate padding
     const dataLength = Buffer.byteLength(JSON.stringify(dataWithPadding), 'utf-8');
-    const modVal = dataLength % DEFAULT_PADDING_LENGTH;
+    const modVal = (dataLength + NACL_EXTRA_BYTES) % DEFAULT_PADDING_LENGTH;
     let padLength = 0;
     // Only pad if necessary
     if (modVal > 0) {
-        padLength = DEFAULT_PADDING_LENGTH - modVal - NACL_EXTRA_BYTES; // nacl extra bytes
+        padLength = DEFAULT_PADDING_LENGTH - modVal;
     }
     dataWithPadding.padding = '0'.repeat(padLength);
     const paddedMessage = JSON.stringify(dataWithPadding);
