diff --git a/src/polfield.js b/src/polfield.js
index 6f570cbe6a431bcfb96ac31bd8dea4b3336ea133..16e1cb11e24348c8cd86e3ed2b931e2760c14ef9 100644
--- a/src/polfield.js
+++ b/src/polfield.js
@@ -191,8 +191,9 @@ export default class PolField {
         }
         return v;
     }
-
-    eval(p,x) {
+    
+    // renamed from eval to make it work in SES
+    eval_sesRename(p,x) {
         const F = this.F;
         if (p.length == 0) return F.zero;
         const m = this._next2Power(p.length);
@@ -223,7 +224,7 @@ export default class PolField {
             let mpol = this.ruffini(roots, points[i][0]);
             const factor =
                 this.F.mul(
-                    this.F.inv(this.eval(mpol, points[i][0])),
+                    this.F.inv(this.eval_sesRename(mpol, points[i][0])),
                     points[i][1]);
             mpol = this.mulScalar(mpol, factor);
             sum = this.add(sum, mpol);
